py/server/deephaven/json/__init__.py

```diff
@@ -0,0 +1,1221 @@
+        allow_missing (bool): if the double value is allowed to be missing, default is True
+        allow_null (bool): if the double value is allowed to be a JSON null type, default is True
+        on_missing (Optional[int]): the value to use when the JSON value is missing and allow_missing is True, default is None.
+        on_null (Optional[int]): the value to use when the JSON value is null and allow_null is True, default is None.
+
+    Returns:
+        the double value
+    """
+    builder = _JDoubleValue.builder()
+    _build(
+        builder,
+        allow_missing,
+        allow_null,
+        allow_decimal=True,
+        allow_int=True,
+        allow_string=allow_string,
+    )
+    if on_null:
+        builder.onNull(on_null)
+    if on_missing:
+        builder.onMissing(on_missing)
+    return JsonValue(builder.build())
+
+
+def string_val(
+    allow_int: bool = False,
+    allow_decimal: bool = False,
+    allow_bool: bool = False,
+    allow_missing: bool = True,
+    allow_null: bool = True,
+    on_missing: Optional[str] = None,
+    on_null: Optional[str] = None,
+) -> JsonValue:
+    """Creates a String value. For example, the JSON string
+
+    .. code-block:: json
+        "Hello, world!"
+
+    might be modelled as the string type
+
+    .. code-block:: python
+        string_val()
+
+    In contexts where the user needs to create a JsonValueType and isn't changing any default values, the user can
+    simplify by using the python built-in str type. For example,
+
+    .. code-block:: python
+        some_method(string_val())
+
+    could be simplified to
+
+    .. code-block:: python
+        some_method(str)
+
+    Args:
+        allow_int (bool): if the string value is allowed to be a JSON integer type, default is False
+        allow_decimal (bool): if the string value is allowed to be a JSON decimal type, default is False
+        allow_bool (bool): if the string value is allowed to be a JSON boolean type, default is False
+        allow_missing (bool): if the double value is allowed to be missing, default is True
+        allow_null (bool): if the double value is allowed to be a JSON null type, default is True
+        on_missing (Optional[str]): the value to use when the JSON value is missing and allow_missing is True, default is None.
+        on_null (Optional[str]): the value to use when the JSON value is null and allow_null is True, default is None.
+
+    Returns:
+        the String value
+    """
+    builder = _JStringValue.builder()
+    _build(
+        builder,
+        allow_missing,
+        allow_null,
+        allow_string=True,
+        allow_int=allow_int,
+        allow_decimal=allow_decimal,
+        allow_bool=allow_bool,
+    )
+    if on_null:
+        builder.onNull(on_null)
+    if on_missing:
+        builder.onMissing(on_missing)
+    return JsonValue(builder.build())
+
+
+# TODO(deephaven-core#5269): Create deephaven.time time-type aliases
```

Comment:
Noting the todo in case anything needs to happen

Reply:
The todo is really meant that the documentation here can be improved once #5269 is merged; right now, I'm relying on `Any`

```python
    on_missing: Optional[Any] = None,
    on_null: Optional[Any] = None,
```

because I don't want to blindly copy over the types from `to_j_instant` as that's fragile (prone to be wrong if `to_j_instant` gets updated).